<a href="https://colab.research.google.com/github/coreyalejandro/Prediction-of-Product-Sales/blob/main/Prediction_of_Product_Sales.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<details>
# <summary>Example</summary>
This is a dropdown with text!
</details>


# Prediction of Product Sales
Usinng Data Science to Identify The Dominant key performance indicators..Product and Place mix opticritica; indicators mal indicators Product and Plaacement Optimal and and Placement Key Product and 

Author: Corey Alejandro

### Assignment Overview

#### Project Overview Part 1

You now have all the skills needed to start your first portfolio project for this course! The goal of your portfolio projects is to have items to show potential employers that demonstrate your data science skills and competencies. You will add to this project over the next few weeks as you learn new skills.  


#### 1. Create a GitHub Account

- Go to [New Window GitHub](https://github.com/) and sign up for an account.

- Once you are signed in, click on your avatar at the top right-hand corner of the page and select "Settings. In Setting:
  * Add a professional profile picture.
  * Add a brief description of who you are or what you are interested in.
  * Optional: include links to your LinkedIn profile, professional Twitter handle, website, or anything else you would want people to find!


---
- Here is an example of what your profile should look like:

![Screen_shot_2021 06_at_10](https://assets.codingdojo.com/boomyeah2015/codingdojo/curriculum/content/chapter/Screen_Shot_2021-01-06_at_10.18.09_AM.png)

---
#### 2. Set up a GitHub repository that will house your first portfolio project.


- To view a tutorial on how to set up GitHub Repository [click here](https://i.vimeocdn.com/video/1036624146-8eea00167e936b56824fd4bded89bca5386abdd1d2b9136ddcfb59894c871d45-d?mw=1500&amp;mh=844&amp;q=70&quot).

- Title your repository “Prediction of Product Sales”




#### 3. Create a Google Colab Notebook with section headers. 

- Create a new notebook. 
- Title your repository “Prediction of Product Sales”
- Add these section headers:
  - H1: Project Title: (Prediction of Product Sales).
  - Author: Include your name in a markdown cell immediately below your header
  - H2: "Project Overview"
  - H2: "Load and Inspect Data
  - H2: "Clean Data"
  - H2: "Exploratory Data Analysis"- H2: "Explanatory Data Analysis"
  
- Here is an example of what your notebook should look like: [Github New Window Project 1 - Part 1](https://github.com/coding-dojo-data-science/ds-fundamentals-notebooks/blob/main/Headers.ipynb)

  
#### 4. Add the Google Colab Notebook to your GitHub repository.

- Open the "file" tab and click on "Save a copy in Github."
- You will be prompted to login to your GitHub.
- Select the repository that you just created for your project.**


#### 5. Submit the link to your GitHub repo!

- Please submit the _**link**_ here, and NOT your notebook file. This will allow your instructor to see that you have been able to successfully set up your GitHub repository and are able to add files to it.




Assignment Overview Part 2

For Part 2, your task is to:
#### 1. Use Pandas to load in the data and start exploring and cleaning it. 
As data scientists, about [New Window 80% of our time is spent on data preparation](https://www.forbes.com/sites/gilpress/2016/03/23/data-preparation-most-time-consuming-least-enjoyable-data-science-task-survey-says/?sh=478ec4796f63) - so this is a very important step!

- Download the data using this link: [New Window download the data](https://drive.google.com/file/d/1syH81TVrbBsdymLT_jl2JIf6IjPXtSQw/view?usp=sharing). (_Note: [New Window Original data source](https://datahack.analyticsvidhya.com/contest/practice-problem-big-mart-sales-iii/))._
- Insert code below the heading "Loading Data"
    
    - Use Pandas to read the sales prediction data set into Google Colab.
    
    - Use df.info() and df.head() methods to preview the first rows of your data and a summary of your DataFrame's columns.
- Insert code below the heading "Data Cleaning"
    
    - Explore the data - what do you need to do to clean this data? Clean it!
    
    - Be sure to address each of the following tasks:

1.  How many rows and columns?

2. What are the datatypes of each variable?

3. Are there duplicates? If so, drop any duplicates.

4. Identify missing values.

5. Decide on how to address the missing values and do it! (This requires your judgement, so explain your choice).

6. Confirm that there are no missing values after addressing them.

7. Find and fix any inconsistent categories of data (example: fix cat, Cat, and cats so that they are consistent).

8. For any numerical columns, obtain the summary statistics of each (min, max, mean).


- Here is the Data Dictionary for this dataset:

<table><tbody><tr><td role="rowheader"><strong>Variable Name</strong></td><td><strong>Description</strong></td></tr><tr><td role="rowheader">Item_Identifier</td><td>Unique product ID</td></tr><tr><td role="rowheader">Item_Weight</td><td>Weight of product</td></tr><tr><td role="rowheader">Item_Fat_Content</td><td>Whether the product is low fat or regular</td></tr><tr><td role="rowheader">Item_Visibility</td><td>The percentage of total display area of all products in a store allocated to the particular product</td></tr><tr><td role="rowheader">Item_Type</td><td>The category to which the product belongs</td></tr><tr><td role="rowheader">Item_MRP</td><td>Maximum Retail Price (list price) of the product</td></tr><tr><td role="rowheader">Outlet_Identifier</td><td>Unique store ID</td></tr><tr><td role="rowheader">Outlet_Establishment_Year</td><td>The year in which store was established</td></tr><tr><td role="rowheader">Outlet_Size</td><td>The size of the store in terms of ground area covered</td></tr><tr><td role="rowheader">Outlet_Location_Type</td><td>The type of area in which the store is located</td></tr><tr><td role="rowheader">Outlet_Type</td><td>Whether the outlet is a grocery store or some sort of supermarket</td></tr><tr><td role="rowheader">Item_Outlet_Sales</td><td>Sales of the product in the particular store. This is the target variable to be predicted.<a href="https://github.com/ShauryaBhandari/Bigmart-Sales-Prediction#why-does-the-the-problem-need-to-be-solved" id="user-content-why-does-the-the-problem-need-to-be-solved" class="anchor" aria-hidden="true" target="_blank" data-acsb-tooltip="New Window" data-acsb-clickable="true" data-acsb-navigable="true" tabindex="-1" data-acsb-now-navigable="false" data-acsb-hidden="true"></a></td></tr></tbody></table>


#### 2. Commit your work to GitHub. 
To do this, you can just select File > Save a Copy in GitHub in your Colab notebook:

![1619113635__screen%20shot%202021 22%20at%2011](https://assets.codingdojo.com/boomyeah2015/codingdojo/curriculum/content/chapter/1619113635__Screen%20Shot%202021-04-22%20at%2011.45.47%20AM.png)

#### 3. Turn in a link to your GitHub repo! 
We will continue to work on the project a little bit each week.

  

Here is an exemplar project which includes an example of Part 2 of the project.

[Github New Window Exemplar Project](https://github.com/coding-dojo-data-science/Project1_Exemplar/blob/main/DS_Salaries.ipynb)

---

## Project Overview

*italicized text*## Loading Data



### Loading Data

Loadoing data refers to the process of importing data from different sources into a data storage system (e.g. database, data warehouse, data lake) in order to be processed, analyzed, and used for various applications

Loading data typically involves the following steps:

1. Identifying the data sources:
2. Extracting the data from its location. 
3. Importing the data into the target system or application.
4. Reading or using it from or within the target system.






---

At Coding Dojo, for this project we:
1. Download the data using this link: [download the data](https://drive.google.com/file/d/1syH81TVrbBsdymLT_jl2JIf6IjPXtSQw/view?usp=sharing). Note: [Original data source](https://datahack.analyticsvidhya.com/contest/practice-problem-big-mart-sales-iii/)
sales prediction data

2.  Save the data file to our Coding Dojo Data Folder in Google Drive.

3. Mount Google Drive to acess to access to access the data.

In [ ]:
# Mount google drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


4. Use Pandas to load , display and cinspect thedatahe data and display it for previewinfg


4. Use Pandas to load, display, and inspect the data in Google Collab Notebooks.4. 

In [ ]:
# Import Pandas and other required librabry packages. 
# Pandas
import pandas as pd
# Seaborn
import seaborn as sns
# MissingNo
import missingno as msno


In [ ]:
# Save the filepath and load the data 
fpath = "/content/drive/MyDrive/CodingDojo/01-Fundamentals/Week02/Data/sales_predictions_2023.csv"
df = pd.read_csv(fpath)

 - Use df.info() and df.head() methods to preview the first rows of your data and a summary of your DataFrame's columns.

In [ ]:
# Display the first 5 rows
df.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

## df(headd) see

In [ ]:
# Display the last 5 rows
df.tail()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
8518           FDF22        6.865          Low Fat         0.056783   
8519           FDS36        8.380          Regular         0.046982   
8520           NCJ29       10.600          Low Fat         0.035186   
8521           FDN46        7.210          Regular         0.145221   
8522           DRG01       14.800          Low Fat         0.044878   

               Item_Type  Item_MRP Outlet_Identifier  \
8518         Snack Foods  214.5218            OUT013   
8519        Baking Goods  108.1570            OUT045   
8520  Health and Hygiene   85.1224            OUT035   
8521         Snack Foods  103.1332            OUT018   
8522         Soft Drinks   75.4670            OUT046   

      Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
8518                       1987        High               Tier 3   
8519                       2002         NaN               Tier 2   
8520                       2004       Small               Tier 2   
8521                       2009      Medium               Tier 3   
8522                       1997       Small               Tier 1   

            Outlet_Type  Item_Outlet_Sales  
8518  Supermarket Type1          2778.3834  
8519  Supermarket Type1           549.2850  
8520  Supermarket Type1          1193.1136  
8521  Supermarket Type2          1845.5976  
8522  Supermarket Type1           765.6700

In [ ]:
# Display the info for the dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


In [ ]:
n use the desiplay

### Cleaning Data

Cleaning data is the process of removing or correcting inaccuracies, inconsistencies, and errors in a dataset to improve its quality and reliability. This involves identifying and addressing missing data, duplicates, outliers, and other issues that may affect the analysis or interpretation of the data. 

The goal of cleaning data is to ensure that the data is consistent, complete, accurate, relevant, and ready for further analysis or use in decision-making. Common techniques for cleaning data include data profiling, data mapping, data standardization, and data validation.s


  After we load and preview the we perform a series of tasks to clean or optimize the tthats daya we scan the .info() to answer some initial questions about the data.Here we wil explore the data to see what we need to do to clean it. Then we clean it! Cleaning is not done in a vaccuum 😉..

.We use a set of targeted questions, along with our understanding of the stakeholder goals, to guide our thinking and decision-making in prepapring the data.


1) How many rows and columns?

-  Columns tell us how many features; rows tell us how many records (also referred to as instances or observations) of the featured information there are.

-  According to the.info(), the data has 
  - 12 columns (features): 
  

5 numeric columns 4 float, 1 integer. These
6 categorical/string columns
  - 8523 rows (records)

- We can verify both the rows and columns by using .shape

In [ ]:
# Verify both with results from .shape (row, col).
df.shape

(8523, 12)

The .info() reported that of the 12 columns, there are
- 5 numeric columns 4 float, 1 integer. These
- 6 categorical/string columns

In [ ]:
df.dtypes

Item_Identifier               object
Item_Weight                  float64
Item_Fat_Content              object
Item_Visibility              float64
Item_Type                     object
Item_MRP                     float64
Outlet_Identifier             object
Outlet_Establishment_Year      int64
Outlet_Size                   object
Outlet_Location_Type          object
Outlet_Type                   object
Item_Outlet_Sales            float64
dtype: object


3) Are there duplicates? If so, drop any duplicates.


```
# This is formatted as code
```



In [ ]:
object_cols = df.select_dtypes(include=['object'])
object_cols

Item_Identifier Item_Fat_Content              Item_Type  \
0              FDA15          Low Fat                  Dairy   
1              DRC01          Regular            Soft Drinks   
2              FDN15          Low Fat                   Meat   
3              FDX07          Regular  Fruits and Vegetables   
4              NCD19          Low Fat              Household   
...              ...              ...                    ...   
8518           FDF22          Low Fat            Snack Foods   
8519           FDS36          Regular           Baking Goods   
8520           NCJ29          Low Fat     Health and Hygiene   
8521           FDN46          Regular            Snack Foods   
8522           DRG01          Low Fat            Soft Drinks   

     Outlet_Identifier Outlet_Size Outlet_Location_Type        Outlet_Type  
0               OUT049      Medium               Tier 1  Supermarket Type1  
1               OUT018      Medium               Tier 3  Supermarket Type2  
2               OUT049      Medium               Tier 1  Supermarket Type1  
3               OUT010         NaN               Tier 3      Grocery Store  
4               OUT013        High               Tier 3  Supermarket Type1  
...                ...         ...                  ...                ...  
8518            OUT013        High               Tier 3  Supermarket Type1  
8519            OUT045         NaN               Tier 2  Supermarket Type1  
8520            OUT035       Small               Tier 2  Supermarket Type1  
8521            OUT018      Medium               Tier 3  Supermarket Type2  
8522            OUT046       Small               Tier 1  Supermarket Type1  

[8523 rows x 7 columns]

How many rows?


1) How many rows and columns?

2) What are the datatypes of each variable?

3) Are there duplicates? If so, drop any duplicates.

4) Identify missing values.

5) Decide on how to address the missing values and do it! (This requires your judgement, so explain your choice).

6) Confirm that there are no missing values after addressing them.

7) Find and fix any inconsistent categories of data (example: fix cat, Cat, and cats so that they are consistent).

8) For any numerical columns, obtain the summary statistics of each (min, max, mean).

Here is the Data Dictionary for this dataset:

Variable Name	Description
Item_Identifier	Unique product ID
Item_Weight	Weight of product
Item_Fat_Content	Whether the product is low fat or regular
Item_Visibility	The percentage of total display area of all products in a store allocated to the particular product
Item_Type	The category to which the product belongs
Item_MRP	Maximum Retail Price (list price) of the product
Outlet_Identifier	Unique store ID
Outlet_Establishment_Year	The year in which store was established
Outlet_Size	The size of the store in terms of ground area covered
Outlet_Location_Type	The type of area in which the store is located
Outlet_Type	Whether the outlet is a grocery store or some sort of supermarket
Item_Outlet_Sales	Sales of the product in the particular store. This is the target variable to be predicted.




# Load the dataset
data = pd.read_csv('train_bm.csv')

# Check the dimensions of the data
print("Original Data Dimensions: ", data.shape)

# Print the data types of each variable
print(df.dtypes)

# Check for duplicates
print("Number of Duplicates: ", 
.duplicated().sum())

# Drop duplicates
df.drop_duplicates(inplace=True)

# Check for missing values in each column

print(df.isnull().sum())

# Fill missing values in 'Item_Weight' column with their median
df['Item_Weight'].fillna(df['Item_Weight'].median(), inplace=True)

# Fill missing values in 'Outlet_Size' column with their mode
df['Outlet_Size'].fillna(df['Outlet_Size'].mode()[0], inplace=True)

# Check for missing values again to confirm they are all filled

df(data.isnull().sum())

# Standardize 'Item_Fat_Content'
df['Item_Fat_Content'] = df['Item_Fat_Content'].replace(['LF', 'reg', 'low fat'], ['Low Fat', 'Regular', 'Low Fat'])

# Check for other inconsistent categories
df("Item_Type Categories:\n", df['Item_Type'].value_counts())

# Summarize the numerical columns

print(data.describe(include=[np.number]))
```
